### PR TITLE
[DowngradePhp74] Handle nested uses on ArrowFunctionToAnonymousFunctionRector

### PR DESCRIPTION
**Pull request by samsonasik**
Closes https://github.com/rectorphp/rector-src/pull/1039

### CODE REVIEW

File: rules/Php72/NodeFactory/AnonymousFunctionFactory.php [hunk 1]
```diff
@@ -187,6 +175,29 @@ public function createAnonymousFunctionFromString(Expr $expr): ?Closure
         return $anonymousFunction;
     }
 
+    private function applyNestedUses(Closure $anonymousFunctionNode, Variable $useVariable): Closure
```
Comment (Lctrs) [inline rules/Php72/NodeFactory/AnonymousFunctionFactory.php:198]: Will it handle 4 or more nested closures ? 🤔

Comment (samsonasik) [reply rules/Php72/NodeFactory/AnonymousFunctionFactory.php:198]: I updated to support 4 nested https://github.com/rectorphp/rector-src/pull/1040/commits/68dc945d4145897b0e5289cc0980aca2ba2b03f7

**Comment by samsonasik** [comment]
@Lctrs this patch is for until 4 nested:

```php
    public function many()
    {
        return fn ($foo) => fn ($bar) => fn ($baz) => fn () => [$foo, $bar, $baz];
    }
```

I think more than that is impractical, unless we can see the real example of the usage of more than that.

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

**Comment by samsonasik** [comment]
@Lctrs I updated to support more than 4 nested, the ClosureUse start to not sorted in 4 nested, but that's ok as it doesn't matter as far as it brought to the next closure https://github.com/rectorphp/rector-src/pull/1040/commits/0e10fc55a449940681ceb734e815d9deb9b96253

**Comment by samsonasik** [comment]
@Lctrs I just updated to keep it sorted in the use closure https://github.com/rectorphp/rector-src/pull/1040/commits/0082aa97e8234b3ae87d7f07dc9b33afa917350f 🎉

**Comment by samsonasik** [comment]
Finally, it support as many nested as needed 🎉🎉🎉, All checks have passed 🎉 @TomasVotruba it is ready for review.

**Comment by Lctrs** [comment]
@samsonasik Following patch add a failing test : 
```diff
diff --git a/rules-tests/DowngradePhp74/Rector/ArrowFunction/ArrowFunctionToAnonymousFunctionRector/Fixture/nested_static_fn_uses.php.inc b/rules-tests/DowngradePhp74/Rector/ArrowFunction/ArrowFunctionToAnonymousFunctionRector/Fixture/nested_static_fn_uses.php.inc
index 61cd83f97..a5d1071ab 100644
--- a/rules-tests/DowngradePhp74/Rector/ArrowFunction/ArrowFunctionToAnonymousFunctionRector/Fixture/nested_static_fn_uses.php.inc
+++ b/rules-tests/DowngradePhp74/Rector/ArrowFunction/ArrowFunctionToAnonymousFunctionRector/Fixture/nested_static_fn_uses.php.inc
@@ -18,6 +18,13 @@ final class NestedStaticFnUses
     {
         return fn ($foo) => fn ($bar) => fn ($baz) => fn ($bat) => fn () => [$foo, $bar, $baz, $bat];
     }
+
+    public function mixed()
+    {
+        return fn($foo) => fn ($bar) => function () use ($foo, $bar) {
+            return [$foo, $bar];
+        };
+    }
 }
 ?>
 -----
@@ -65,5 +72,16 @@ final class NestedStaticFnUses
             };
         };
     }
+
+    public function mixed()
+    {
+        return function ($foo) {
+            return function ($bar) use ($foo) {
+                return function () use ($foo, $bar) {
+                    return [$foo, $bar];
+                };
+            };
+        };
+    }
 }
 ?>
```

The `$foo` variable is added to the use list of the last closure but it's already there.

**Comment by samsonasik** [comment]
@Lctrs Fixed 🎉 https://github.com/rectorphp/rector-src/pull/1040/commits/6483982030eeb41c4d10ad55738c5331d6826ebd

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

**Comment by samsonasik** [comment]
I am merging it ;)

**Pull request merged**